pkg/network/usm/sharedlibraries/watcher_test.go

```diff
@@ -458,7 +438,7 @@ func (s *SharedLibrarySuite) TestSoWatcherProcessAlreadyHoldingReferences() {
 	require.GreaterOrEqual(t, tel["usm.so_watcher.hits"], tel["usm.so_watcher.matches"], "usm.so_watcher.hits")
 	telEqual(t, 1, "usm.so_watcher.already_registered")
 	telEqual(t, 0, "usm.so_watcher.blocked")
-	telEqual(t, 3, "usm.so_watcher.matches") // command1 access to 2 files, command2 access to 1 file
+	telEqual(t, 0, "usm.so_watcher.matches")
```

Comment:
how that can be possible?

Reply:
This is an artifact of the testing code being slightly improved and actually triggering the initialization code of the library which was untested until this point.
The `so_watcher_client` program now actually mmaps the files, similiarly to what the GNU linker (ld) will actually do when loading a shared library.

The reason why we have `usm.so_watcher.matches` set to zero now is because this metric gets only incremented for shared libraries loaded  at runtime, and not shared libraries already in memory and matched during initialization